modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -6184,8 +6185,41 @@ public void testMoveObjectEntryToTrash() throws Exception {
 		AssertUtils.assertEquals(
 			WorkflowConstants.STATUS_IN_TRASH, objectEntryVersion.getStatus());
 
+		AssertUtils.assertFailure(
+			ObjectEntryVersionStatusException.CannotExpireEntryInTrash.class,
+			"Cannot expire a version of an entry that is in recycle bin",
+			() -> _defaultObjectEntryManager.expireObjectEntryByVersion(
+				dtoConverterContext, _objectDefinition1, objectEntryId, 1));
+
+		_updateObjectEntryVersion(_objectDefinition1, objectEntry1, 2);
```

Comment:
The entry was already deleted (moved to trash), are you trying to generate a new version for this entry?